### PR TITLE
Fix Stepper performance bug

### LINKED ISSUE #97337: Stepper: Improve performances
https://github.com/Automattic/wp-calypso/issues/96926#issuecomment-2533163210

By @MicBosi 

I also ran a few tests 4X for /start/user-setup and 4X for /setup/onboarding/user and one thing that captured my attention was the Largest Contentful Paint figure.

For Start I have: 10s, 10.2s, 10.4s, 9.6s
For Stepper I have: 17.9s, 16.1s, 13.1s, 13.1s

Granted these numbers are a bit volatile but they match another manual test I had done refreshing Stepper and Start in Chrome, using 3G and 4G Slow Network Throttling (to accentuate the differences) and tracking how many seconds it takes for the various social options to appear on the. page:

3G/Start: ~27s (~3.2MB downloaded)
3G/Stepper: 36s (~4.5MB downloaded)

I would say these are just preliminary figures but it might be worth looking into some of these key performance metrics to make sure Stepper is as optimised as Start before the next A/B test.

### PR DESCRIPTION
**Pull request by alshakero**
In another instance of tree-shaking breaking down, Stepper was pulling a ton of JS and CSS that it never runs since https://github.com/Automattic/wp-calypso/pull/67887.

Fixes https://github.com/Automattic/wp-calypso/issues/97337

## Proposed Changes

This moves the `setupErrorLogger` function from a big file full of imports into its own file to help tree shaking.

## Why are these changes being made?
To make Stepper faster. 

## Testing Instructions

1. Go to /setup/onboarding.
2. Make sure dashicons.css is not loaded. 

### Regression testing
~~Follow the testing steps of https://github.com/Automattic/wp-calypso/pull/67887~~
These steps don't pass on prod.

### CODE REVIEW

File: client/landing/stepper/index.tsx [hunk 1]
```diff
@@ -18,8 +18,8 @@ import { createRoot } from 'react-dom/client';
 import { Provider } from 'react-redux';
 import { BrowserRouter, matchPath } from 'react-router-dom';
 import { requestAllBlogsAccess } from 'wpcom-proxy-request';
-import { setupErrorLogger } from 'calypso/boot/common';
 import { setupLocale } from 'calypso/boot/locale';
+import { setupErrorLogger } from 'calypso/boot/setup-error-logger';
```
Comment (jsnajdr) [inline client/landing/stepper/index.tsx]: A better location for this module would be `calypso/lib/error-logger`. The `boot` folder should contain only the minimal boot code and import most stuff from elsewhere. Let's not repeat the mistake when we stuffed the entire Stepper framework into `client/landing` 🙂

**Review by jsnajdr — APPROVED** [review]
What a nice catch 🙂 Simple and effective. I just have one suggestion about the location of the `setupErrorLogger` code.

**Review by tyxla — COMMENTED** [review]
Thanks for the great fix @alshakero, really nice catch ❤️

**Comment by alshakero** [comment]
Thank you sir 🙇🏼

**Comment by MicBosi** [comment]
Fantastic @alshakero! 🎉 🎉 🎉

**Comment by sirbrillig** [comment]
This is still under discussion in Slack (p1734034864133699-slack-C0BNMNMNG) , but for the record it looks like this PR unintentionally caused some weird style bugs appear in various parts of calypso. So far we've noticed display bugs on `/start/domains` and `/me/purchases` where previously a div was `display: flex` and now it's `display: block`.

**Pull request merged**